src/store/modules/ModDownloadModule.ts

```diff
@@ -0,0 +1,112 @@
+import {ActionTree, GetterTree} from 'vuex';
+
+import {ImmutableProfile} from '../../model/Profile';
+import ThunderstoreCombo from '../../model/ThunderstoreCombo';
+import ThunderstoreDownloaderProvider from '../../providers/ror2/downloading/ThunderstoreDownloaderProvider';
+import {State as RootState} from '../index';
+import R2Error from "../../model/errors/R2Error";
+
+// TODO: Replace the "enum" in '../../model/enums/StatusEnum' with a proper enum like this one.
+enum StatusEnum {
+    FAILURE = 0,
+    PENDING = 1,
+    SUCCESS = 2,
+}
+
+interface ProgressItem {
+    modName: string;
+    installProgress: number;
+    downloadProgress: number;
+    status: number; // StatusEnum
+    error: R2Error | null;
+}
+
+interface State {
+    downloads: ProgressItem[];
+}
+
+/**
+ * State for Mod Downloading/Installation.
+ */
+export default {
+    namespaced: true,
+
+    state: (): State => ({
+        downloads: [],
+    }),
+
+    getters: <GetterTree<State, RootState>>{
+        activeDownload(state): ProgressItem | undefined {
+            return state.downloads.slice(-1)[0]; // Last element of the array
+        },
+    },
+    mutations: {
+        reset(state: State) {
+            state.downloads = [];
+        },
+        addDownload(state: State, modName: string) {
+            state.downloads.push({
+                modName: modName,
+                installProgress: 0,
+                downloadProgress: 0,
+                status: StatusEnum.PENDING,
+                error: null
+            });
+        },
+        updateDownloadProgress(state: State, params: { progress: number, modName: string, status: number, err: R2Error | null }) {
+            let downloadMod = state.downloads.find((progressItem) => progressItem.modName === params.modName);
+
+            // Doesn't do anything when the downloaded mod is not found in the array (which means it is a dependency).
```

Comment:
Seem dangerous to assume it's always a dependency instead of just a bug. Also based on gut feeling I think we're going to need track all the combos in the progress item, not just the "main download". This includes dependencies when downloading a single mod, and multiple mods with their dependencies when updating all installed mods to latest versions.